ios/FluentUI/Navigation/SideBar.swift

```diff
@@ -0,0 +1,284 @@
+
+		NSLayoutConstraint.activate([
+			leadingAnchor.constraint(equalTo: superView.leadingAnchor),
+			topAnchor.constraint(equalTo: superView.topAnchor),
+			bottomAnchor.constraint(equalTo: superView.bottomAnchor)
+		])
+	}
+
+	private let backgroundView: UIVisualEffectView = {
+		var style = UIBlurEffect.Style.regular
+		if #available(iOS 13, *) {
+			style = .systemMaterial
+		}
+
+		return UIVisualEffectView(effect: UIBlurEffect(style: style))
+	}()
+
+	private let borderLine = Separator(style: .shadow, orientation: .vertical)
+
+	private func initialize() {
+		translatesAutoresizingMaskIntoConstraints = false
+
+		backgroundView.translatesAutoresizingMaskIntoConstraints = false
+		contain(view: backgroundView)
+
+		borderLine.translatesAutoresizingMaskIntoConstraints = false
+		addSubview(borderLine)
+
+		addSubview(topStackView)
+		addSubview(bottomStackView)
+
+		if #available(iOS 13, *) {
+			addInteraction(UILargeContentViewerInteraction())
+		}
+
+		accessibilityTraits = .tabBar
+
+		NSLayoutConstraint.activate([widthAnchor.constraint(equalToConstant: Constants.viewWidth),
+									 borderLine.leadingAnchor.constraint(equalTo: trailingAnchor),
+									 borderLine.bottomAnchor.constraint(equalTo: bottomAnchor),
+									 borderLine.topAnchor.constraint(equalTo: topAnchor)])
+	}
+
+	private var layoutConstraints: [NSLayoutConstraint] = []
+
+	private func setupLayout() {
+		if layoutConstraints.count > 0 {
+			NSLayoutConstraint.deactivate(layoutConstraints)
+			layoutConstraints.removeAll()
+		}
+
+		if let avatarView = self.avatarView {
+			layoutConstraints.append(contentsOf: [
+				avatarView.topAnchor.constraint(equalTo: safeAreaLayoutGuide.topAnchor, constant: 18.0),
+				avatarView.centerXAnchor.constraint(equalTo: centerXAnchor),
+				avatarView.widthAnchor.constraint(equalToConstant: Constants.avatarViewSize),
+				avatarView.heightAnchor.constraint(equalToConstant: Constants.avatarViewSize),
+				topStackView.topAnchor.constraint(equalTo:avatarView.bottomAnchor , constant: 34.0)
+			])
+		} else {
+			layoutConstraints.append(topStackView.topAnchor.constraint(equalTo: safeAreaLayoutGuide.topAnchor, constant: 30.0))
+		}
+
+		layoutConstraints.append(contentsOf: [
+			topStackView.widthAnchor.constraint(equalTo: widthAnchor),
+			topStackView.heightAnchor.constraint(equalToConstant: stackViewHeight(topStackView)),
+			bottomStackView.widthAnchor.constraint(equalTo: widthAnchor),
+			bottomStackView.heightAnchor.constraint(equalToConstant: stackViewHeight(bottomStackView)),
+			bottomStackView.bottomAnchor.constraint(equalTo: safeAreaLayoutGuide.bottomAnchor, constant: -10.0)
+		])
+
+		NSLayoutConstraint.activate(layoutConstraints)
+	}
+
+	private func stackViewHeight(_ stackView: UIStackView) -> CGFloat {
+		let itemCount: CGFloat = CGFloat(stackView.arrangedSubviews.count)
+		var height: CGFloat = 24.0 * itemCount
+
+		if (itemCount > 0) {
+			height += stackView.spacing * (itemCount - 1)
+		}
+
+		return height
+	}
```

Comment:
Can we instead set the contentCompressionResistance/hugging priorities as appropriate on the items within the stack view so it automatically takes its height

Reply:
Unfortunately, the intrinsic size of TabBarItemView is zero height and zero width. TabBarItemView contains an image view and a label. Those take up space and are outside of the view's bounds. This definitely seems like a bug. I tried fixing TabBarItemView so that the root view contains all its subviews but this caused severe layout bugs in TabBarView. I think fixing TabBarItemView's behavior is out of scope for this change. What is the process to file a bug?
If TabBarItemView's view was sized correctly, we wouldn't need logic at all. Currently, our stack view thinks that TabBarItemView have a zero size.